lib/sidekiq/job_metadata_middleware.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+module Sidekiq
+  class JobMetadataMiddleware
+    def call(job_instance, job_payload, _queue)
+      return unless job_instance.is_a?(Sidekiq::JobMetadata)
+
+      job_instance.instance_variable_set(:@job_metadata, job_payload)
```

Comment:
🤔 would it make sense to add accessors to set `job_metadata` rather than metaprogram through to it?

Reply:
Yep, I think that's a great idea to make it more readable. I pushed an update.